src/main/java/DemandeAvenantComponent.java

```diff
@@ -0,0 +1,433 @@
+import fr.aba.prevoyance.postgresql.entity.DemandeAvenantContratEntity;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import fr.aba.prevoyance.api.service.DemandeAvenantService;
+import com.aviva.prevoyance.pivot.adhesion.contrat.avenant.DemandeAvenantPivot;
+import com.aviva.prevoyance.pivot.adhesion.contrat.v2.ContratPivot;
+import com.aviva.prevoyance.pivot.couverture.v2.CouverturePivot;
+import com.aviva.prevoyance.pivot.couverture.v2.GarantiePivot;
+import com.aviva.prevoyance.pivot.finance.PrimePivot;
+import com.aviva.prevoyance.pivot.finance.SurprimePivot;
+import com.aviva.prevoyance.pivot.personne.v2.RolePersonnePivot;
+import com.aviva.prevoyance.pivot.v2.DossierPivot;
+import com.aviva.prevoyance.pivotparametrage.IntervalleMontantsPivot;
+import com.aviva.prevoyance.pivotparametrage.avenant.DemandeAvenantConstantes;
+import com.aviva.prevoyance.pivotparametrage.avenant.DemandeAvenantConstantes.EtatDemandeAvenant;
+import com.aviva.prevoyance.pivottechnical.exceptions.v2.ComponentCoherenceDatasException;
+import com.aviva.prevoyance.pivottechnical.exceptions.v2.ComponentValidationException;
+import com.aviva.prevoyance.pivottechnical.exceptions.v2.EnumNotFoundException;
+import fr.aba.prevoyance.postgresql.entity.DemandeAvenantEntity;
+import fr.aba.prevoyance.postgresql.repository.DemandeAvenantContratRepository;
+import fr.aba.prevoyance.postgresql.repository.DemandeAvenantRepository;
+import com.aviva.prevoyance.utils.component.BigDecimalComponent;
+import com.aviva.prevoyance.utilsmicroservice.component.contrat.APIGarantie;
+
+@Component
+public class DemandeAvenantComponent {
+
+	@Autowired private DemandeAvenantRepository repositoryDemandeAvenant;
+	@Autowired private DemandeAvenantContratRepository repositoryDemandeAvenantContrat;
+	@Autowired private DemandeAvenantConverter convDemandeAvenant;
+	@Autowired private DemandeAvenantService sDemandeAvenant;
+	@Autowired private APIGarantie apiGarantie;
+
+	@Autowired private GarantieComponent cGarantie;
+	@Autowired private PersonneComponent cPersonne;
+	@Autowired private SurprimesComponent surprimesComponent;
+	@Autowired private ContratConverter convContrat;
+	@Autowired private BigDecimalComponent cBigDecimal;
+
+	public Optional<DemandeAvenantPivot> demandeOuvertureNouvelleDemandeAvenantV2(DossierPivot dossier, UUID uuidTrace) {
+		BigInteger idDemandeAvenant = sDemandeAvenant.createNewDemandeAvenantV2(dossier, uuidTrace);
+		return getDemandeAvenant(idDemandeAvenant);
+	}
+
+	private Predicate<? super CouverturePivot> nePasPrendreLesCouverturesRefusees() {
+		return couverture -> !StringUtils.equals("5", couverture.getEtat().getCodeEtat());
+	}
+
+	public Optional<DemandeAvenantPivot> createDemandeAvenantAcceptee(DossierPivot dossier, BigInteger idDemandeAvenant, BigInteger idDemandeAvenantPhoto, UUID uuidTrace) {
+		// Obtenir les garanties du contrat
```

Comment:
Potential SQL injection vulnerability due to direct concatenation of user input in the query. Input validation and parameterized queries should be used to prevent SQL injection. (Severity: high)